packages/calypso-e2e/src/lib/pages/published/published-post-page.ts

```diff
@@ -91,4 +94,15 @@ export class PublishedPostPage extends BaseContainer {
 		await this._click( selectors.unlikeButton );
 		await this.frame.waitForSelector( selectors.notLikedText, { state: 'visible' } );
 	}
+
+	/**
+	 * Confirms the presenced of the block on the published page.
+	 *
+	 * This method accepts a selector referencing the block to verify whether the block(s) are rendered on page.
+	 *
+	 * @param {string} blockSelector Selector to block(s).
+	 */
+	async confirmBlockPresence( blockSelector: string ): Promise< void > {
+		await this.page.waitForSelector( `${ selectors.postBody } ${ blockSelector }` );
+	}
```

Comment:
This should be a static method of an individual block, not a general, block-unspecific one. Blocks can differ in a way we'd like them to be verified on the published page - i.e. Image block should also perform a check that the image has been rendered. For example, if I'd like to verify the presence of the `ClickToTweet` block, I'd rather see it called via `await ClickToTweet.validatePublishedContent()` that would already include the `.wp-block-coblocks-click-to-tweet` class, as well as the checks for tweeted text, etc. Does that make sense?

Reply:
That sounds like a good approach.

Reply:
I've implemented a static method in each instance of the block to perform checks.

What I'd like to know is if there's a way to generalize this some more. Currently, each block needs to have its own `static validatePublishedContent()` method and this can be problematic if we need to change how the function operates. Do you have any ideas to improve the current implementation?

Reply:
I think this is OK since each block is different so it requires a different validation. For those checks, I'd focus more on the content that the block produces than on checking if the wrapper element is present. For example, for the `ClickToTweet` block, I think it would bring more value if we confirm that the tweet's text is present:

```js
static async validatePublishedContent( page,  tweet ) {
	await page.waitForSelector( `text=${ tweet }` );
}
```

This way we kinda don't need to even use the block's selector as the text check is sufficient for validating that the block produced expected content.

Similarly, with the `PricingTable` block, it would be great to validate the actual content we published.

What do you think?

Reply:
I'm in two minds about that, personally.

On one hand, validating the presence of the selectors ought to be enough that the block is present. This covers situations like with the Pricing Table block where if pricing data was not entered the block selector itself would not show up on the published page. Bonus, this keeps the `validatePublishedContent` signature consistent across implementations and manageable.

On the other hand checking for the content to be present would definitely help with ensuring the appropriate values are shown on the block after publishing. With my short time at a8c so far I haven't run into a situation like this, but I'm sure it can happen with a new block or significantly updated block. This would potentially make the method signature very long and complex if the list of things to validate on grows:

```
validatePublishedContent( page, price1, price2, feature1, feature2, plan1name, plan2name, ...)
```

Reply:
With all that I've typed up, I ended up modifying `validatePublishedContent` to check for values. Please take a look and let me know if it should be changed.

Reply:
Because this function seems to be intended to be used for polymorphism (in the looping test), do you think it's worth it to try to collapse the signatures down to a single signature and include that as part of an interface that each Block implements? That way, tests in TypeScript that consume these blocks could key off that interface and have more strict type checking. 

It would need to no longer be static, but I think that could end up fine. We could also, for each block, define two POM classes per module, one to represent the block in the editor, and another to represent the block in a published page. The former could take the block `ElementHandle` as its dependency in its constructor, and the latter could take the `page` in its constructor. The latter would be what would implement the interface with this method.

Reply:
> We could also, for each block, define two POM classes per module, one to represent the block in the editor, and another to represent the block in a published page.

Interesting idea and worth exploring some more. It would provide better separation of concerns and method calls tailored for each use case.

My concern would be around overhead for both developers implementing a new block in this module as well as test writers. I would imagine test writers needing to context switch between the editor and published representations of the block:

```
PricingTableBlock.something

...

PricingTableBlockPublished.validate()
```

whereas currently the test writer only needs to know that `PricingTableBlock` can handle both use cases.

Reply:
>  do you think it's worth it to try to collapse the signatures down to a single signature and include that as part of an interface that each Block implements?

Would this work for blocks that require no validation eg. Dynamic HR block? The validation method there only checks whether the class exists on page.

Reply:
Personally, I think we should hold off on making this change for now and perhaps look at revising this when additional block tests are migrated. It should not be too difficult to make the changes at that time.

Reply:
Sounds good to me! We can definitely revisit this.